src/kernel_common.cpp

```diff
@@ -2,41 +2,6 @@
 
 namespace SCAMP {
 
-template <typename T>
```

Comment:
I actually don't think this needs to be a template at all. We can leave it the same as it was but remove the template parameter. Does that fix the problem?